docker/transport/npipesocket.py

```diff
@@ -131,22 +133,38 @@ def recv_into(self, buf, nbytes=0):
         if not isinstance(buf, memoryview):
             readbuf = memoryview(buf)
 
+        event = win32event.CreateEvent(None, True, True, None)
+        overlapped = pywintypes.OVERLAPPED()
+        overlapped.hEvent = event
         err, data = win32file.ReadFile(
             self._handle,
-            readbuf[:nbytes] if nbytes else readbuf
+            readbuf[:nbytes] if nbytes else readbuf,
+            overlapped
         )
-        return len(data)
+        wait_result = win32event.WaitForSingleObject(event, self._timeout)
+        if wait_result == win32event.WAIT_TIMEOUT:
+            win32file.CancelIo(self._handle)
+            raise TimeoutError
+        return win32file.GetOverlappedResult(self._handle, overlapped, 0)
 
     def _recv_into_py2(self, buf, nbytes):
+        raise RuntimeError
         err, data = win32file.ReadFile(self._handle, nbytes or len(buf))
         n = len(data)
         buf[:n] = data
         return n
 
```

Comment:
```suggestion
```

I just double-checked, this used to be gated by a `six` call, must have been missed when the rest of the Python2 code went away, but it can be removed entirely now

Reply:
alright, I removed the function entirely

Reply:
Thanks for doing the extra clean-up ✨